### DISCUSSION TITLE
+loading.svelte component for every route.

### DISCUSSION THREAD

**Discussion by Devr-pro** [sveltejs/kit#8288] [Ideas]
Hello there, this discussion might not be new but want to share some thoughts about +loading.svelte component for every route folder, by this way we can render a loading component for a pending state , like when `load ` function is still running,etc this will be so efficient for apps and developers too! Thanks. (I am a noob so I might not know how this will be working 😅)

**Comment by EmilTholin** [comment]
Hi @Devr-pro!

This would be nice indeed. [This feature is tracked here](https://github.com/sveltejs/kit/issues/7213).

In the meantime you might be able to use the [`navigating` store from `$app/stores`](https://kit.svelte.dev/docs/modules#$app-stores-navigating):

```svelte
<script>
  import { navigating } from '$app/stores';
</script>

{#​if $navigating}
  <div>Loading...</div>
{/if}
```

**Reply by Devr-pro** [reply]
Oh thanks for that issue link, and yes am currently using navigating store as the solution.
